pubnub-core/src/subscription/subscribe_loop.rs

```diff
@@ -0,0 +1,282 @@
+use super::encoded_channels_list::EncodedChannelsList;
+use super::registry::{RegistrationEffect, Registry as GenericRegistry, UnregistrationEffect};
+use super::subscribe_request;
+use crate::message::Timetoken;
+use crate::transport::Transport;
+use futures_channel::{mpsc, oneshot};
+use futures_util::future::{select, Either, FutureExt};
+use futures_util::sink::SinkExt;
+use futures_util::stream::StreamExt;
+use log::{debug, error};
+
+pub(crate) use super::channel::{Rx as ChannelRx, Tx as ChannelTx};
+pub(crate) use super::registry::ID as SubscriptionID;
+
+pub(crate) type Registry = GenericRegistry<ChannelTx>;
+
+pub(crate) type ReadyTx = oneshot::Sender<()>;
+#[allow(dead_code)]
+pub(crate) type ReadyRx = oneshot::Receiver<()>;
```

Comment:
I would prefer not allowing dead code. Let's not maintain this. YAGNI.

Reply:
I agree. The reason I left those like that is cause I was considering slicing the crate into the smaller chunks (making those parts of the public API), but ended up not doing that.